packages/compass-components/src/components/document-list/document-edit-actions-footer.tsx

```diff
@@ -0,0 +1,343 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import type { default as HadronDocumentType } from 'hadron-document';
+import { Element } from 'hadron-document';
+import { Button } from '../leafygreen';
+import { css, spacing, uiColors } from '../..';
+
+const enum Status {
+  Initial = 'Initial',
+  Editing = 'Editing',
+  Modified = 'Modified',
+  ContainsErrors = 'ContainsErrors',
+  UpdateStart = 'UpdateStart',
+  UpdateBlocked = 'UpdateBlocked',
+  UpdateSuccess = 'UpdateSuccess',
+  UpdateError = 'UpdateError',
+  Deleting = 'Deleting',
+  DeleteStart = 'DeleteStart',
+  DeleteSuccess = 'DeleteSuccess',
+  DeleteError = 'DeleteError',
+}
+
+function isDeleting(status: Status): boolean {
+  return [
+    Status.Deleting,
+    Status.DeleteStart,
+    Status.DeleteSuccess,
+    Status.DeleteError,
+  ].includes(status);
+}
+
+function isSuccess(status: Status): boolean {
+  return [Status.DeleteSuccess, Status.UpdateSuccess].includes(status);
+}
+
+function isPrimaryActionDisabled(status: Status): boolean {
+  return [
+    Status.Editing,
+    Status.ContainsErrors,
+    Status.UpdateStart,
+    Status.UpdateSuccess,
+    Status.DeleteStart,
+    Status.DeleteSuccess,
+  ].includes(status);
+}
+
+function isCancelDisabled(status: Status): boolean {
+  return [
+    Status.UpdateStart,
+    Status.UpdateSuccess,
+    Status.DeleteStart,
+    Status.DeleteSuccess,
+  ].includes(status);
+}
+
+const StatusMessages: Record<Status, string> = {
+  [Status.Initial]: '',
+  [Status.Editing]: '',
+  [Status.Deleting]: 'Document flagged for deletion.',
+  [Status.Modified]: 'Document modified.',
+  [Status.ContainsErrors]:
+    'Update not permitted while document contains errors.',
+  [Status.UpdateStart]: 'Updating document...',
```

Comment:
[tiny nit, but since this something we display:]

```suggestion
  [Status.UpdateStart]: 'Updating document…',
```